metabase/metabase
# Fix drag + drop operations on collections pages

Closes https://github.com/metabase/metabase/issues/30614

### Description

Basically, `react-dropzone` was either swallowing all drop events on the collections page, or our other libraries were swallowing them all. This composes the event handlers from `react-dropzone` to only handle file drop events and allows all our drag + drop operations to play nicely together.

![ALLTHEDRAGS](https://github.com/metabase/metabase/assets/30528226/ff523dd3-874c-4fff-9afb-d2ad0e62cb2d)


### How to verify

Testing:
- enable uploads in admin settings
- visit any collection
- see that you can upload files by dragging them into the main collection item list area 🥳 
- see that you can drag items around in pinned items as well 🥳 

### Checklist

The event handler code is unit tested, but I think it might be impossible to e2e test this drag behavior with cypress currently. I'll ping nemanja to see if he has any ideas.



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30718)
